### ISSUE TITLE
Question: application_resume, application_paused or application_state?

### ISSUE THREAD

**Issue by jbeemster** [snowplow/iglu-central#362]
In the Android Tracker we can trace the Activity being resumed or paused (foreground and background state of the application).  Should this be split into two distinct schemas or nested into a single application_state context whereby we have:

```json
{
   "type" : "Activity",
   "isForeground": "true | false"
}
```

This tells us what type of screen was backgrounded (can be expanded to include other non-Android types) and the state of the application.  In combination with the client_session you can start to view how many changes occurred during a session etc.

Thoughts @alexanderdean ?

**Comment by alexanderdean** [comment]
Hey @jbeemster - can you clarify what the event(s) are here? Ideally in a subject-verb-object format... I think that will help to model this.

**Comment by jbeemster** [comment]
So here I was saying that we can either have:

1. Event for an application resuming (either initial startup or being resumed from the background).  The first of this event to fire will be flagged as the startup event.
2. Event for an application stopping (going to background)
3. Combining both of these events into one and naming it as `application_state` or to use more Android specific lingo `application_lifecycle` - as this event is essentially hooking into two of the lifecycle states.

```json
{
  "type" : "Activity", // The type of screen we are on
  "isStartup" : "true | false", // If this was the first state event to be fired true, else false
  "isForeground" : "true | false" // If the application is in the fore or background
}
```

This would be triggered of of a listener that waits for the foreground state to change and then fires.

**Comment by alexanderdean** [comment]
Hey @jbeemster - it sounds to me like we have two events:

* `application_foreground`
* `application_background`

What I am not clear on:

1. What does a screen of type "Activity" have to do with the app foregrounding or backgrounding? They feel orthogonal - what am I missing?
2. With the `isStartup` flag for `application_foreground`, is this something you are deriving somehow (if so, how), or is it a part of the `application_lifecycle`?

**Comment by jbeemster** [comment]
1. You are right these are completely orthogonal.  I guess I was just trying to add context to where they were in the application when these events occurred ... perhaps adding something like the name of the Activity and or Fragment here would be better to give us a glimpse into where a user left?

2. The `isStartup` would be flagged on the first `application_foreground` event.  This is because in the Android lifecycle the `onCreate()` function can be called many times so it is not a good indicator of whether an app was just started - but we can note this on the first occurrence.

**Comment by alexanderdean** [comment]
Hey @jbeemster -

> perhaps adding something like the name of the Activity and or Fragment here would be better to give us a glimpse into where a user left?

Could this be a context?

> The isStartup would be flagged on the first application_foreground event

Why not just keep and send an index of each foregrounding and backgrounding?

**Comment by jbeemster** [comment]
Yeah could that as well for the isStartup.  True that could be quite a nice context!
